superset-frontend/src/features/home/ChartTable.test.tsx

```diff
@@ -68,64 +69,53 @@ describe('ChartTable', () => {
     showThumbnails: false,
   };
 
-  let wrapper: ReactWrapper;
+  const otherTabProps = {
+    ...mockedProps,
+    otherTabData: mockCharts,
+  };
+
+  const mineTabProps = {
+    ...mockedProps,
+    mine: mockCharts,
+  };
+
+  const renderOptions = {
+    useRedux: true,
+    useRouter: true,
+  };
 
-  beforeEach(async () => {
-    act(() => {
-      wrapper = mount(<ChartTable store={store} {...mockedProps} />);
+  const renderChartTable = (props: any) =>
+    act(async () => {
+      render(<ChartTable {...props} />, renderOptions);
     });
-    await waitForComponentToPaint(wrapper);
-  });
 
-  it('renders', () => {
-    expect(wrapper.find(ChartTable)).toExist();
+  it('renders with EmptyState if no data present', async () => {
+    await renderChartTable(mockedProps);
+    expect(screen.getAllByRole('tab')).toHaveLength(3);
+    expect(
+      screen.getByText(/other charts will appear here/i),
+    ).toBeInTheDocument();
   });
 
   it('fetches chart favorites and renders chart cards', async () => {
-    act(() => {
-      const handler = wrapper.find('[role="tab"] a').at(0).prop('onClick');
-      if (handler) {
-        handler({} as any);
-      }
+    await renderChartTable(mockedProps);
+    fireEvent.click(screen.getByText(/favorite/i));
```

Comment:
Let's use `userEvent` instead

Reply:
@justinpark curious what should be the go-to usage, `fireEvent` or `userEvent`, [as you mentioned](https://github.com/apache/superset/pull/27422/files/c9930915bd2be345a4af772953f2995b5b4545f4#r1516655143) we should save `userEvent` for special cases but this seems to go against the [RTL wiki](https://github.com/apache/superset/wiki/Testing-Guidelines-and-Best-Practices#react-testing-library-rtl) and [docs](https://testing-library.com/docs/user-event/intro#differences-from-fireevent)

Reply:
@justinpark sent [this article ](https://stackoverflow.com/questions/61080116/ant-design-v4-breaks-react-testing-library-tests-for-select-and-autocomplete). Switching to `userEvent`